### PR TITLE
fix(groq): Feature/groq response format improvements

### PR DESCRIPTION
**Pull request by allohamora**
This PR updates the langchain groq response_format type by replacing it with one from the groq-sdk and adds the example from #6708 to the documentation.

### CODE REVIEW

**Comment by jacoblee93** [comment]
Thanks, this looks good!

**Review by jacoblee93 — APPROVED** [review]

File: docs/core_docs/docs/integrations/chat/groq.ipynb [hunk 1]
```diff
@@ -103,7 +103,10 @@
     "    maxTokens: undefined,\n",
     "    maxRetries: 2,\n",
     "    // other params...\n",
-    "})"
+    "})\n",
+    "\n",
+    "// if you need a response format other than the default\n",
+    "const llmWithResponseFormat = llm.bind({ response_format: \"json_object\" });"
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/chat/groq.ipynb:109]: Or wait, does this need to be `response_format: { type: "json_object" }`? I think that's what OpenAI does

Comment (allohamora) [reply docs/core_docs/docs/integrations/chat/groq.ipynb:109]: Yes
image

**Comment by allohamora** [comment]
> Thanks, this looks good!
> 
> BTW you should also be able to pass it in at runtime if you don't want to use `.bind()`:
> 
> ```ts
> await model.invoke(messages, { response_format: "json_object" });
> ```

I have added a Json invocation block with 2 ways to pass response_format here https://github.com/langchain-ai/langchainjs/pull/6754/commits/5f0c6a75fd8e5e52544151444beae778008b1b0e

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**